### DISCUSSION TITLE
Google Analytics 4 Support

### DISCUSSION THREAD

**Discussion by daniel-keller** [thegetty/quire#284] [Development]
Quire relies on Google's Universal Analytics. Universal has been deprecated and will be sunset on July 1, 2023. Quire will need to switch to GA4. My proposal is to use 2 analytics configuration settings to apply either Universal or GA4 until 2023. 
Populating the config setting `GoogleAnalyticsGA4: "G-12345"` would load an alternative  analytics js file into the html head.

For example (not tested):
```
<!-- Google Analytics GA4 -->
<script>
(function(i, s, o, g, r, a, m) {
      i['GoogleAnalyticsObject'] = r;
      i[r] = i[r] || function() {
          (i[r].q = i[r].q || []).push(arguments)
      }, i[r].l = 1 * new Date();
      a = s.createElement(o),
          m = s.getElementsByTagName(o)[0];
      a.async = 1;
      a.src = g;
      m.parentNode.insertBefore(a, m)
  })(window, document, 'script', 'https://www.googletagmanager.com/gtag/js?id={{ .Site.GoogleAnalyticsGA4 }}', 'ga');

  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}

  gtag('js', new Date());
  gtag('config', '{{ .Site.GoogleAnalyticsGA4 }}');

  document.addEventListener('DOMContentLoaded', function() {
    var links = document.querySelectorAll("a[href$='pdf'],a[href$='rtf'],a[href$='doc'],a[href$='xls'],a[href*='bit.ly'],a[href$='csv'],a[href$='json'],a[href$='zip'],a[href$='ppt'],a[href*='epub'],a[href*='mobi']")
    var link, pdfLabel, pdfOnClick;
    for (var i = 0; i < links.length; i++) {
      link = links[i];
      pdfLabel = link.getAttribute('href');
      pdfOnClick = "gtag('event', 'page_view', {page_title: '" + pdfLabel + "'});";
      link.setAttribute("onClick", pdfOnClick);
    }
  }, { once:true });
</script>

```

**Comment by szeygoh** [comment]
Hi, can anyone who has experience on this share if Quire now support GA4?

**Comment by daniel-keller** [comment]
I'm not familiar with the latest version of Quire running on llty but from what I see in the [code](https://github.com/thegetty/quire/blob/024f4cfffd495abbb733bae584ef1bebe548e5bd/packages/11ty/_includes/components/analytics.js#L24), quire is still injecting analytics.js which I believe is only for Universal and not gtag.js which GA4 requires. @geealbers is this correct?

**Reply by szeygoh** [reply]
Thank you!

**Comment by Erin-Cecele** [comment]
Please see PR #1035 for in progress work on GA4 Analytics
